codegen/smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ServiceClientConfigGeneratorTest.kt

```diff
@@ -255,85 +253,20 @@ public class Config private constructor(builder: Builder) {
-
-        /**
-         * The policy to use for evaluating operation results and determining whether/how to retry.
-         */
-        override var retryPolicy: RetryPolicy<Any?>? = null
-
-        /**
-         * The [RetryStrategy] implementation to use for service calls. All API calls will be wrapped by the
-         * strategy.
-         */
-        override var retryStrategy: RetryStrategy? = null
-
-        /**
-         * Configure events that will be logged. By default clients will not output
-         * raw requests or responses. Use this setting to opt-in to additional debug logging.
-         *
-         * This can be used to configure logging of requests, responses, retries, etc of SDK clients.
-         *
-         * **NOTE**: Logging of raw requests or responses may leak sensitive information! It may also have
-         * performance considerations when dumping the request/response body. This is primarily a tool for
-         * debug purposes.
-         */
-        override var sdkLogMode: SdkLogMode = SdkLogMode.LogRequest
-
-        /**
-         * The tracer that is responsible for creating trace spans and wiring them up to a tracing backend (e.g.,
-         * a trace probe). By default, this will create a standard tracer that uses the service name for the root
-         * trace span and delegates to a logging trace probe (i.e.,
-         * `DefaultTracer(LoggingTraceProbe, "<service-name>")`).
-         */
-        override var tracer: Tracer? = null
-"""
-        contents.shouldContainWithDiff(expectedBuilderProps)
+        // Expect sdkLogMode config value to override default to SdkLogMode.Request
+        val expectedConfigValues = """
+    override val clientName: String = builder.clientName
+    override val httpClientEngine: HttpClientEngine = builder.httpClientEngine ?: DefaultHttpEngine().manage()
+    override val authSchemes: kotlin.collections.List<aws.smithy.kotlin.runtime.http.auth.HttpAuthScheme> = builder.authSchemes
+    public val customProp: Int? = builder.customProp
+    public val endpointProvider: EndpointProvider = requireNotNull(builder.endpointProvider) { "endpointProvider is a required configuration property" }
+    override val idempotencyTokenProvider: IdempotencyTokenProvider = builder.idempotencyTokenProvider ?: IdempotencyTokenProvider.Default
+    override val interceptors: kotlin.collections.List<aws.smithy.kotlin.runtime.http.interceptors.HttpInterceptor> = builder.interceptors
+    override val retryPolicy: RetryPolicy<Any?> = builder.retryPolicy ?: StandardRetryPolicy.Default
+    override val retryStrategy: RetryStrategy = builder.retryStrategy ?: StandardRetryStrategy()
+    override val sdkLogMode: SdkLogMode = builder.sdkLogMode ?: SdkLogMode.LogRequest
+    override val tracer: Tracer = builder.tracer ?: DefaultTracer(LoggingTraceProbe, clientName)"""
```

Comment:
Since I changed the `SdkLogMode` config property to use `ConfigPropertyType.RequiredWithDefault` instead of defining a `defaultValue` on the `symbol`, the builder field now always defaults to `null`

I tried playing around with this test-case for a while and I ended up with this modification